alpha_vantage/alphavantage.py

```diff
@@ -281,90 +377,56 @@ def _format_wrapper(self, *args, **kwargs):
-            json_response = response.json()
-            if not json_response:
-                raise ValueError(
-                    'Error getting data from the api, no return was given.')
-            elif "Error Message" in json_response:
-                raise ValueError(json_response["Error Message"])
-            elif "Information" in json_response and self.treat_info_as_error:
-                raise ValueError(json_response["Information"])
-            elif "Note" in json_response and self.treat_info_as_error:
-                raise ValueError(json_response["Note"])
-            return json_response
-        else:
-            csv_response = csv.reader(response.text.splitlines())
-            if not csv_response:
-                raise ValueError(
-                    'Error getting data from the api, no return was given.')
-            return csv_response
+                        # JSON to CSV conversion through Pandas.
+                        # Convert to CSV reader to stay consistent with native CSV API call.
+                        return csv.reader(data_pandas.to_csv().splitlines()), meta_data
+            elif 'csv' == api_format:
+                if 'pandas' == output_format or \
+                        'json' == output_format:
+                    # Converts the CSV reader to a format that can be read by pandas
+                    # directly
+                    data_list = list(call_response)
+                    data_pandas = pandas.DataFrame.from_records(data_list[1:],
+                                                                columns=data_list[0])
+                    
+                    data_pandas = cls._convert_pandas_index(self, data_pandas)
+                    
+                    if 'pandas' == output_format:
+                        return data_pandas, None
+                    else:
+                        return data_pandas.to_json(), None
+                else:
+                    return call_response, None
+        
+        def _convert_pandas_index(self, av, data_pandas: pandas.DataFrame):
+            """Converts the index of a pandas.DataFrame as specified in the AlphaVantage class."""
+            if not isinstance(data_pandas.index, pandas.RangeIndex) and \
+                    'integer' == av.indexing_type and 'default' == self.formatting:
+                # Set Date/Time as an actual column so a new numerical index
+                # will be created, but only when specified by the user.
+                data_pandas.reset_index(level=0, inplace=True)
+                data_pandas.rename(columns={'index': 'date'}, inplace=True)
+                data_pandas.index.name = 'index'
+            elif not isinstance(data_pandas.index, pandas.DatetimeIndex) and \
+                    'date' == av.indexing_type and 'default' == self.formatting:
+                # TODO check if index is actually a date(time) and only convert if this is the case
+                if 'time' in data_pandas.columns:
```

Comment:
this wont work for most of the api calls, not limited but including: TIME_SERIES_DAILY, TIME_SERIES_DAILY_ADJUSTED.
Because the column isnt named **"time"** , for the api calls above it would be named **"timestamp"**.

Reply:
That is absolutely right.

Adding another check for the `timestamp` column should resolve this issue.

I implemented this while taking the result of `TIME_SERIES_INTRADAY_EXTENDED` as reference. It provides a `time` column. I'd be happy if you take another look at the new commit.